File: Programming_Ex/C/Collatz.c
#include <stdio.h>
#include <stdlib.h>

int on_even(int num);
int on_odd(int num);

int main(int argc, char** argv)
{
	int num = 0;
	int i = 0;
	for(i=1; i<argc; i++)
	{
		num = atoi(argv[i]);
		if(!num || num < 0 || num != atof(argv[i]))
		{
			printf("%s is not a valid number.\n",argv[i]);
		}
		else
		{
			printf("%d\n",num);
			while(num != 1)
			{
				if(num % 2 == 0)
				{
					num = on_even(num);
				}
				else
				{
					num = on_odd(num);
				}
				printf("\t%d\n",num);
			}
		}
	}
	return 0;
}

int on_even(int num)
{
	return num / 2;
}

int on_odd(int num)
{
	return num * 3 + 1;
}

File: Programming_Ex/PHP/Collatz.php
<?php
	//The Collatz Conjecture
	$nums = explode(',',$_POST["numbers"]);
	Collatz($nums);
	/*
	echo $_POST["numbers"]."<br/>";
	foreach($nums as $num)
	{
		echo $num;
	}
	echo "<br/>";
	*/

	function on_even($num)
	{
		return $num / 2;
	}

	function on_odd($num)
	{
		return $num * 3 + 1;
	}

	function Collatz($nums)
	{
		for($i=0; $i<count($nums); $i++)
		{
			$num = intval($nums[$i]);
			if($num == 0 || $num < 0 || $num != floatval($nums[$i]))
			{
				echo $nums[$i] . " is not a valid number.<br/>";
			}
			else
			{
				echo $num . "<br/>";
				while($num != 1)
				{
					if($num % 2 == 0)
					{
						$num = on_even($num);
					}
					else
					{
						$num = on_odd($num);
					}
					echo $num . "<br/>";
				}
			}
			echo "---------------------------<br/>";
		}
	}
?>
